include/aspect/geometry_model/chunk.h

```diff
@@ -0,0 +1,205 @@
+   * from the +ve x-axis, as per the mathematical convention of phi.
+   * In 3D, the class simulates a chunk of a sphere, bounded by arbitrary
+   * lines of longitude, latitude and radius. Boundary indicator names are
+   * west, east, south, north, inner and outer.
+   *
+   * The parameters that describe this geometry and that are read from the
+   * input file are the inner and outer radii of the shell, the minimum
+   * and maximum longitude, minimum and maximum longitude, and the
+   * number of cells initialised in each dimension.
+   */
+
+  template <int dim>
+  class ChunkGeometry : public ChartManifold<dim,dim>
+  {
+    public:
+      virtual
+      Point<dim>
+      pull_back(const Point<dim> &space_point) const;
+
+      virtual
+      Point<dim>
+      push_forward(const Point<dim> &chart_point) const;
+  };
+
+}
+
+namespace aspect
+{
+  namespace GeometryModel
+  {
+    using namespace dealii;
+
+    /**
+     * A class that describes a piece of a circle/sphere delimited by
+     * minimum and maximum longitude and radius (2D), or
+     * minimum and maximum longitude, latitude and radius (3D)
+     */
+    template <int dim>
+    class Chunk : public Interface<dim>
+    {
+      public:
+
+        /**
+         * Generate a coarse mesh for the geometry described by this class.
+         */
+        virtual
+        void create_coarse_mesh (parallel::distributed::Triangulation<dim> &coarse_grid) const;
+
+        /**
+         * Return the set of boundary indicators that are used by this model.
+         * This information is used to determine what boundary indicators can
+         * be used in the input file.
+         *
+         * The box model uses boundary indicators zero through 2*dim-1, with
+         * the first two being the faces perpendicular to the x-axis, the next
+         * two perpendicular to the y-axis, etc.
+         */
+        virtual
+        std::set<types::boundary_id>
+        get_used_boundary_indicators () const;
+
+        /**
+         * Return a mapping from symbolic names of each part of the boundary
+         * to the corresponding boundary indicator. This allows users to
+         * specify *names*, not just *numbers* in their input files when
+         * describing which parts of the boundary have to satisfy which
+         * boundary conditions.
+         *
+         * This geometry returns the map <code>{{"left"->0}, {"right"->1},
+         * {"bottom"->2}, {"top"->3}}</code> in 2d, and <code>{{"left"->0},
+         * {"right"->1}, {"front"->2}, {"back"->3}, {"bottom"->4},
+         * {"top"->5}}</code> in 3d.
+         */
+        virtual
+        std::map<std::string,types::boundary_id>
+        get_symbolic_boundary_names_map () const;
+
+
+        /**
+         * Return the typical length scale one would expect of features in
+         * this geometry, assuming realistic parameters.
+         *
+         * We return 1/100th of the diameter of the box.
```

Comment:
Here you use 1/100th of the box, but the function actually implements the spherical shell scale. Please sync comment and function